source/CommonJobs/CommonJobs.MVC.UI/Scripts/Evaluations/admin.js

```diff
@@ -70,12 +78,63 @@
             CurrentPosition: this.currentPosition,
             Seniority: this.seniority,
             FullName: this.fullName,
-            Period: this.period
+            Period: this.period,
+            IsValid: this.isValid()
         };
     }
-
     var viewmodel = new PeriodCreation();
 
+    $('#generate-evaluation-button').on('click', function () {
+        if (viewmodel.isValid()) {
+            postEvaluationsForGeneration({
+                onSuccess: function (result) {
+                    showEvaluationGenerationResult(true, result.amountGenerated);
+                },
+                onError: function () {
+                    alert('Fallo interno. Por favor recargue la página.');
+                }
+            });
+        } else {
+            showEvaluationGenerationResult(false);
+        }
+
+        function postEvaluationsForGeneration(callbacks) {
```

Comment:
You should not declare this function inside the other anonymous function. This way, you're re-generating this function each time that the other one is invoked. I think you should take it out of the `$('#generate-evaluation-button').on` invocation.